src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -380,33 +450,119 @@ def _convert_templates_in_files(
+        metrics.set_counter_default(CLICounterField.TEMPLATES_PROCESSOR, 0)
+        artifacts_to_template = [
+            artifact
+            for artifact in pkg_model.artifacts
+            for processor in artifact.processors
+            if processor.name == TEMPLATES_PROCESSOR
+        ]
+        if not in_memory and artifacts_to_template:
+            metrics.set_counter(CLICounterField.TEMPLATES_PROCESSOR, 1)
+
+            # Create a temporary directory to hold the expanded templates,
+            # as if a bundle step had been run but without affecting any
+            # files on disk outside of the artifacts we want to convert
+            with tempfile.TemporaryDirectory() as d:
+                deploy_root = Path(d)
+                bundle_ctx = BundleContext(
+                    package_name=pkg_model.identifier,
+                    artifacts=pkg_model.artifacts,
+                    project_root=project_root,
+                    bundle_root=project_root / pkg_model.bundle_root,
+                    deploy_root=deploy_root,
+                    generated_root=(
+                        project_root / deploy_root / pkg_model.generated_root
+                    ),
+                )
+                template_processor = TemplatesProcessor(bundle_ctx)
+                for artifact in artifacts_to_template:
+                    bundle_map = BundleMap(
+                        project_root=project_root, deploy_root=deploy_root
+                    )
+                    bundle_map.add(artifact)
+
+                    for src, dest in bundle_map.all_mappings(
+                        absolute=True, expand_directories=False
+                    ):
+                        # Copy the mapping from the source root to the deploy root,
+                        # since the processor expects the artifacts to have
+                        # already been bundled (we can't call build_bundle() since it
+                        # checks that the deploy_root is a child of the project_root,
+                        # which isn't the case here)
+                        symlink_or_copy(src, dest, deploy_root=deploy_root)
+
+                    for src, dest in bundle_map.all_mappings(
+                        absolute=True, expand_directories=True
+                    ):
+                        if src.is_dir():
+                            continue
+                        # We call the implementation directly instead of calling process()
+                        # since we need access to the BundleMap to copy files anyways
+                        template_processor.expand_templates_in_file(
+                            src, dest, replacement_template_context
+                        )
+                        # Copy the expanded file back to its original source location if it was modified
+                        if not dest.is_symlink():
+                            shutil.copyfile(dest, src)
```

Comment:
we always want to copy the destination back to the source, but if it's a symlink we can't because it's the same file, so we only need to do it if it's not a symlink. If we stop using symlinks, the `if` check will pass and the copy will still happen